app/services/statistics/base.rb

```diff
@@ -18,16 +20,22 @@ def id
       @id || count_by.to_s
     end
 
-    def url
-      @url || "statistics/#{id}"
-    end
-
     def search
       "/hosts?search=#{@search}"
     end
 
     def metadata
       {:id => id, :title => title, :url => url, :search => search}
     end
+
+    private
+
+    def build_url
+      taxonomies = {}
+      taxonomies[:location_id] = @location_id if @location_id
+      taxonomies[:organization_id] = @organization_id if @organization_id
+
+      Rails.application.routes.url_helpers.url_for({:controller => 'statistics', :action => 'show', :id => id, :only_path => true}.merge(taxonomies))
```

Comment:
this should be `:id => @id` I think

Reply:
Why? if there's no id set explicitly, it should fallback `count_by`, if you want I can refactor the method so we do `@id = options[:id] || count_by.to_s` to initializer so `id` would remain pure reader. But it seems unrelated and I'm trying to keep this small so it can be backported.

Reply:
right, missed that. technically i would prefer that be done in the initiallizer but I won't hold this PR on that.